### PR TITLE
add support for vulkan-shader-profiler external passes

### PR DESCRIPTION
**Pull request by rjodinchr**
Depends on https://github.com/KhronosGroup/SPIRV-Headers/pull/398

### CODE REVIEW

File: source/table.h [hunk 1]
```diff
@@ -74,7 +74,7 @@ typedef struct spv_ext_inst_desc_t {
   const uint32_t ext_inst;
   const uint32_t numCapabilities;
   const spv::Capability* capabilities;
-  const spv_operand_type_t operandTypes[16];  // TODO: Smaller/larger?
+  const spv_operand_type_t operandTypes[64];  // TODO: Smaller/larger?
```
Comment (dneto0) [inline source/table.h:77]: I think the corresponding extended instruction set only needs 33 slots.
I suggest lowering this to 40.

The fixed array sizes are kind of ugly, a quick implementation tradefoff I made about 7 years go.  I'm long overdue making a more compact but still static and immutable data structure.  That's on me.

Comment (rjodinchr) [reply source/table.h]: I have lowered it to 40

**Comment by rjodinchr** [comment]
Also, please note that this PR introduces a dependency of `spirv-tools` on `vulkan-headers`

**Comment by dneto0** [comment]
I haven't looked deeply.  Requiring a depdency on vulkan-headers seems unfortunate.

**Comment by rjodinchr** [comment]
I have removed the dependency on vulkan-headers.
I will keep the passes in another repository, but those changes are still needed for those to work

**Review by dneto0 — APPROVED** [review]

**Review by dneto0 — APPROVED** [review]

**Review by dneto0 — APPROVED** [review]

**Pull request merged**